test/unit_tests/braket/experimental/autoqasm/test_api.py

```diff
@@ -917,3 +920,39 @@ def main():
 
     with pytest.raises(errors.AutoQasmTypeError):
         main()
+
+
+def test_gate_calibrations():
+    """test multiple gate calibrations"""
+    gate_calibrations = GateCalibrations()
+
+    @gate_calibrations.register("h", (0,))
+    def cal_1():
+        pulse.barrier(0)
+
+    @gate_calibrations.register("h", (1,))
+    def cal_2():
+        pulse.delay(1, 0.123)
```

Comment:
This pattern is fine on its own but it is a little bit disjointed with the rest of the `@aq.` decorators.

What about a pattern like:
```
@aq.calibration("h", (0,))
def cal_1(): ...

@aq.calibration("h", (1,))
def cal_2(): ...

gate_calibrations = GateCalibrations(cal_1, cal_2)

@aq.main
def my_program(): ...

device = LocalSimulator()
device.run(my_program(), gate_calibrations=gate_calibrations)
```